python/ray/train/xgboost/xgboost_trainer.py

```diff
@@ -1,21 +1,34 @@
 import logging
 from functools import partial
-from typing import Any, Dict, Optional
+from typing import Any, Callable, Dict, Optional, Union
 
 import xgboost
 from packaging.version import Version
 
 import ray.train
 from ray.train import Checkpoint
-from ray.train.constants import _DEPRECATED_VALUE, TRAIN_DATASET_KEY
+from ray.train.constants import TRAIN_DATASET_KEY
 from ray.train.trainer import GenDataset
-from ray.train.xgboost import RayTrainReportCallback
+from ray.train.xgboost import RayTrainReportCallback, XGBoostConfig
 from ray.train.xgboost.v2 import XGBoostTrainer as SimpleXGBoostTrainer
 from ray.util.annotations import PublicAPI
 
 logger = logging.getLogger(__name__)
 
 
+LEGACY_XGBOOST_TRAINER_DEPRECATION_MESSAGE = (
+    "The legacy XGBoostTrainer API is deprecated in favor of the new "
+    "XGBoostTrainer API which accepts a ``train_loop_per_worker`` argument, "
+    "similar to the other DataParallelTrainer APIs (ex: TorchTrainer). "
+    "This gives more flexibility in how users want to train their XGBoost models, "
+    "compared to the legacy API which is a restrictive, opaque wrapper API "
+    "that only allows training models via the ``xgboost.train()`` function "
+    "and must be constantly updated to support new XGBoost features. "
+    "See this issue for more context: "
+    "https://github.com/ray-project/ray/issues/50042"
+)
```

Comment:
nit: Might be unclear for users what the "legacy" XGBoostTrainer API is. Better to explicitly just call out what is deprecated (passing in `label_column`, `params`, `num_boost_round`, `train_kwargs`), in favor of `train_loop_per_worker`. Any rationale or additional detail can be in the github issue.